packages/block-editor/src/hooks/style.js

```diff
@@ -246,9 +246,7 @@ export const withBlockControls = createHigherOrderComponent(
 const withElementsStyles = createHigherOrderComponent(
 	( BlockListBlock ) => ( props ) => {
 		const elements = props.attributes.style?.elements;
-		if ( ! elements ) {
-			return <BlockListBlock { ...props } />;
```

Comment:
Yes, this causes a remount which is not great for `BlockEdit` components. We had similar issues in other hooks in the past. So yeah the fix here is looking good to me.